snooty/main.py

```diff
@@ -284,5 +288,5 @@ def main() -> None:
             print("Closing connection...")
             connection.close()
 
-    if args["build"] and backend.total_warnings > 0:
-        sys.exit(1)
+    if args["build"] and backend.total_errors > 0:
+        sys.exit(EXIT_STATUS_ERROR_DIAGNOSTICS)
```

Comment:
So what is the difference between exit code 1 and 2 here? How does returning exit code 2 make it so the autobuilder can catch unexpected parser terminations?

Reply:
Exit code 1 indicates unexpected termination, while 2 indicates that the parse finished, but with errors the writer should definitely look at. If error code 1 is returned, then the autobuilder should not run the frontend at all.